src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -308,15 +314,54 @@ class ContentSidebar extends PureComponent<Props, State> {
      * @param {Object} file - Box file
      * @return {void}
      */
-    fetchFileSuccessCallback = (file: BoxItem): void => {
-        if (SidebarUtils.shouldRenderSidebar(this.props, file)) {
-            this.setState({
+    fetchMetadataSuccessCallback = (
+        file: BoxItem,
+        editors?: Array<MetadataEditor>,
+    ): void => {
+        let newState = { isVisible: false };
+        if (SidebarUtils.shouldRenderSidebar(this.props, file, editors)) {
+            newState = {
                 file,
+                editors,
                 isVisible: true,
-                view: this.getDefaultSidebarView(file, this.props),
-            });
+                view: this.getDefaultSidebarView(this.props, file, editors),
+            };
+        }
+        this.setState(newState);
+    };
+
+    /**
+     * File fetch success callback that sets the file and view
+     * Only set file if there is data to show in the sidebar.
+     * Skills sidebar doesn't show when there is no data.
+     *
+     * @private
+     * @param {Object} file - Box file
+     * @return {void}
+     */
+    fetchFileSuccessCallback = (file: BoxItem): void => {
+        const { metadataSidebarProps }: Props = this.props;
+        const {
+            getMetadata,
+            isFeatureEnabled = true,
```

Comment:
The naming here is unclear given that we're in ContentSidebar. Which feature is this referring to? Should the default value live in `defaultProps`?